### PR TITLE
Select multiple environment configs

### PR DESCRIPTION
**Pull request by P0t4T0o**
### Description of your changes

related to: https://github.com/crossplane/crossplane/issues/3770, see https://github.com/crossplane/crossplane/issues/3770#issuecomment-1491628182

with introduced changes composite will aggregate all environment configs which were selected by labels

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

via unit tests & in local environment

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by MisterMX — APPROVED** [review]

File: internal/controller/apiextensions/composite/environment/selector.go [hunk 1]
```diff
@@ -106,28 +112,33 @@ func (s *APIEnvironmentSelector) buildEnvironmentConfigRefFromRef(ref *v1.Enviro
...
 		matchLabels[m.Key] = val
 	}
 	res := &v1alpha1.EnvironmentConfigList{}
 	if err := s.kube.List(ctx, res, matchLabels); err != nil {
-		return corev1.ObjectReference{}, errors.Wrap(err, errListEnvironmentConfigs)
+		return []corev1.ObjectReference{}, errors.Wrap(err, errListEnvironmentConfigs)
 	}
 	if len(res.Items) == 0 {
-		return corev1.ObjectReference{}, errors.New(errListEnvironmentConfigsNoResult)
+		return []corev1.ObjectReference{}, errors.New(errListEnvironmentConfigsNoResult)
```
Comment (bobh66) [inline internal/controller/apiextensions/composite/environment/selector.go:129]: I know it's technically not part of this change, but it seems like it should be valid for a Selector to return no results, just as it is valid for a CompositionSelector to return no results and fallback to the defaultComposition.  In this case the environment may be constructed of a variety of named and selected EnvironmentConfigs, as well as patches from the Composite, and the absence of one or more selectors may not be a problem, and may even be expected.

**Comment by P0t4T0o** [comment]
> Does this also happen to address #3941 too? 🙏

can I include changes mentioned here: https://github.com/crossplane/crossplane/issues/3941#issuecomment-1499203688 in this PR or do you want to introduce policy? that could be done in separate PR

**Comment by P0t4T0o** [comment]
@MisterMX @jbw976 
I created new PR: https://github.com/crossplane/crossplane/pull/3981 which addresses aggregation of all env configs with matching labels and introduces resolve policy. Could you please review changes? Will then close this PR in favour of the new one

**Comment by phisco** [comment]
This was actually superseded by https://github.com/crossplane/crossplane/pull/3981, which was merged.

**Pull request closed without merging**